lib/state-store.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * StateStore stores state and allows an observer to subscribe to state updates
+ */
+export default class StateStore {
+  constructor() {
+    this._listeners = [];
+    this._state = {};
+  }
+
+  get state() {
+    return this._state;
+  }
+
+  update(props) {
+    Object.assign(this._state, props);
+    this._listeners.forEach(listener => listener());
```

Comment:
This is a very coarse-grained subscription. Do you intend for subscribers to receive the message and then re-query their state through the controller API?

Reply:
Yeah the idea is the view simply listens for an update and requeries the state as part of its vdom render.  The only use of this is the wire the view’s render. It’s intended to be a dumb callback.